extensions/amp-experiment/0.1/test/test-variant.js

```diff
@@ -167,6 +177,28 @@ describe('allocateVariant', () => {
     })).to.eventually.equal(null);
   });
 
+  it('allow variant override from URL fragment', () => {
+    getParamStub.withArgs('amp-x-name').returns('-Variant_1');
+    return expect(allocateVariant(fakeWin, 'name', {
+      sticky: false,
+      variants: {
+        '-Variant_1': 50,
+        '-Variant_2': 50,
+      },
+    })).to.eventually.equal('-Variant_1');
+  });
+
+  it('variant override should ignore non-existed variant name', () => {
+    getParamStub.withArgs('amp-x-name').returns('-Variant_3');
+    return expect(allocateVariant(fakeWin, 'name', {
+      sticky: false,
+      variants: {
+        '-Variant_1': 50,
+        '-Variant_2': 50,
+      },
+    })).to.eventually.equal('-Variant_2');
+  });
+
```

Comment:
nit: a test for case-sensitivity would be nice here to protect against `getParam` suddenly becoming case-insensitive and breaking this. (there are some web frameworks that lower-case QS params).